api/controllers/v0.3/ILSHelper.js

```diff
@@ -0,0 +1,390 @@
+    return `https://${process.env["ILS_SERVER"]}/iii/patronio/services/PatronIO?wsdl`;
+  }
+
+  callAndFailGracefully(params) {
+    try {
+      this.client.call(params);
+    } catch (e) {
+      let error;
+      switch (
+        e["fault"]["detail"]["com_iii_webapps_ws_patronio_patron_io_fault"][
+          "code"
+        ]
+      ) {
+        case IlsHelper.NOT_FOUND:
+          error = "NotFoundError()";
+          break;
+        case IlsHelper.MULTIPLE_MATCHES:
+          error = "MultipleMatchesError()";
+          break;
+        default:
+          error = "IlsError()";
+      }
+      throw new Error(error);
+      // TODO: catch HTTP timout errors
+      // These errors are returned when the SOAP client times out
+      // when connecting and when requesting (respectively).
+    }
+  }
+
+  search(query) {
+    return this.callAndFailGracefully({
+      type: "search_patrons",
+      message: {
+        search_string: query,
+        username: process.env["ILS_USERNAME"],
+        password: process.env["ILS_PASSWORD"],
+      },
+    });
+  }
+
+  create_patron(patron) {
+    if (!patron.validForIls) {
+      throw new Error("IlsError");
+    }
+    let ilsPatron = this.patronFields(patron);
+
+    return this.callAndFailGracefully({
+      type: "create_patron",
+      message: {
+        patron: ilsPatron,
+        username: process.env["ILS_USERNAME"],
+        password: process.env["ILS_PASSWORD"],
+      },
+    });
+  }
+
+  updatePatron(patron) {
+    if (!patron.validForIls) {
+      throw new Error("IlsError");
+    }
+    let ilsPatron = this.patronFields(patron);
+
+    return this.callAndFailGracefully({
+      type: "update_patron",
+      message: {
+        patron: ilsPatron,
+        username: process.env["ILS_USERNAME"],
+        password: process.env["ILS_PASSWORD"],
+      },
+    });
+  }
+
+  getPatronIdFromResponse(response) {
+    return response.body["multi_ref"][0]["patronId"];
+  }
+
+  available(barcodeOrUsername) {
+    const result = search(
+      `//${IlsHelper.BARCODE_FIELDTAG}//${barcodeOrUsername}`
+    );
+    return !!(result === "NotFoundError");
+  }
+
+  // Format a patron's address as ILS expects it.
+  formatAddress(address) {
+    return address.toUpperCase.gsub(/[\n\r]+/, "$");
+  }
+
+  // Internal: Format card fields for ILS patron creation/updates.
+  formattedPatronData(patron) {
+    let email;
+    let birthdate;
+    let workAddress;
+    if (patron.email) {
+      email = patron.email.gsub(/\s+/, "").toUpperCase;
+    }
+    if (patron.birthdate) {
+      // birthdate = patron.birthdate.strftime("%Y%m%d000000");
+      birthdate = patron.birthdate;
+    }
+
+    let address = this.formatAddress(patron.address);
+    if (patron.worksInCity) {
+      workAddress = this.formatAddress(patron.workAddress);
+    }
+
+    let fullName = patron.name.toUpperCase();
+    if (!fullName.includes(",")) {
+      // Existing TODO: Replace this code with a call to a dedicated human name
```

Comment:
There doesn't seem to be a branch for what happens if the name does include a comma.  (Also, why would someone's name ever include a comma??)

Reply:
Maybe the user added it by mistake or the ILS returned "Guzman, Edwin"... not sure. Most of this file has the existing logic so I haven't really touched most of the branching logic. That's in the next PR.